### PR TITLE
refactored validators for config to allow the validator to change the value

### LINKED ISSUE #280: Implement duration config suffixes
For duration configs (e.g. ELASTIC_APM_API_REQUEST_TIME), we will support time units like 1ms, 1s, 1m. For pre-v2 config, we should allow suffixes, but also continue to allow no suffix and assume the previous unit (e.g. "ms" for min span duration). For v2+ config, we should require suffixes.

### PR DESCRIPTION
**Pull request by beniwohli**
this allows implementation of e.g. unit parsers in the validator. A
UnitValidator that makes use of this is also included

closes #280

### CODE REVIEW

File: tests/config/tests.py [hunk 1]
```diff
@@ -131,3 +141,42 @@ class MyConfig(_ConfigBase):
...
+
+    c = MyConfig({"BYTE": "10b", "KBYTE": "5kib", "MBYTE": "17mib", "WRONG_PATTERN": "5 kb"})
+    assert c.byte == 10
+    assert c.kbyte == 5 * 1024
+    assert c.mbyte == 17 * 1024 * 1024
+    assert c.wrong_pattern is None
+    assert "WRONG_PATTERN" in c.errors
+
+
+def test_duration_validation():
```
Comment (graphaelli) [inline tests/config/tests.py]: Need a test for multiple validators?

Comment (beniwohli) [reply tests/config/tests.py]: good idea, added

**Comment by beniwohli** [comment]
Thanks for the review @graphaelli, great input as always!

I didn't really realize until now that renaming `span_frames_min_duration_ms` is backwards incompatible. That kinda sucks :/ We'll have to do a major version bump anyway for v2 and distributed tracing, but I hoped I could merge this before. Guess that's not going to happen then. I'll merge it into the v2-api branch.

**Comment by beniwohli** [comment]
Merged into #281, closing this one

**Pull request closed without merging**